test/kubectx.bats

```diff
@@ -87,3 +104,81 @@ load common
   echo "$(get_context)"
   [[ "$(get_context)" = "user2@cluster1" ]]
 }
+
+@test "get one context and switch to non existent context" {
+  use_config config1
+
+  run ${COMMAND} "unknown-context"
+  echo "$output"
+  [ "$status" -eq 1 ]
+}
+
+@test "rename context" {
+  use_config config2
+
+  run ${COMMAND} "new-context=user1@cluster1"
+  echo "$output"
+  [ "$status" -eq 0 ]
+
+  run ${COMMAND}
+  echo "$output"
+  [ "$status" -eq 0 ]
+  [[ "$output" = *"new-context"* ]]
+  [[ "$output" = *"user2@cluster1"* ]]
+}
+
+@test "rename current context" {
+  use_config config2
+
+  run ${COMMAND} user2@cluster1
+  echo "$output"
+  [ "$status" -eq 0 ]
+
+  run ${COMMAND} new-context=.
+  echo "$output"
+  [ "$status" -eq 0 ]
+
+  run ${COMMAND}
+  echo "$output"
+  [ "$status" -eq 0 ]
+  [[ "$output" = *"user1@cluster1"* ]]
+  [[ "$output" = *"new-context"* ]]
+}
+
+@test "delete context" {
+  use_config config2
+
+  run ${COMMAND} -d "user1@cluster1"
+  echo "$output"
+  [ "$status" -eq 0 ]
+
+  run ${COMMAND}
+  echo "$output"
+  [ "$status" -eq 0 ]
+  [[ "$output" = "user2@cluster1" ]]
+}
+
+@test "delete current context" {
+  use_config config2
+
+  run ${COMMAND} user2@cluster1
+  echo "$output"
+  [ "$status" -eq 0 ]
+
+  run ${COMMAND} -d .
+  echo "$output"
+  [ "$status" -eq 0 ]
+
+  run ${COMMAND}
+  echo "$output"
+  [ "$status" -eq 0 ]
+  [[ "$output" = "user1@cluster1" ]]
```

Comment:
instead of testing "what's left in the output" (fragile, because we can change what's in `config2`), test "what shouldn't be in the output".

so please test the deleted context no longer shows up in the output. syntax is: `[[ ! "$output = *"name"* ]]`